src/Spice86.Core/Emulator/Memory/MainMemory.cs

```diff
@@ -0,0 +1,59 @@
+namespace Spice86.Core.Emulator.Memory; 
+
+using Spice86.Core.Emulator.VM;
+
+/// <summary>
+/// Represents the main memory of the IBM PC.
+/// Size must be at least 1 MB.
+/// </summary>
+public class MainMemory : Memory {
+    /// <summary>
+    /// Size of conventional memory in bytes.
+    /// </summary>
+    public const uint ConvMemorySize = 1024 * 1024;
+
+    private readonly Machine _machine;
+    
+    public MainMemory(Machine machine, uint sizeInKb) : base(sizeInKb) {
+        _machine = machine;
+        if (sizeInKb * 1024 < ConvMemorySize) {
+            throw new ArgumentException("Memory size must be at least 1 MB.");
+        }
+    }
+    
+    public override void SetUint32(uint address, uint value) {
+        _machine.Ems?.TryWriteMappedPageData(address, value);
+        base.SetUint32(address, value);
```

Comment:
Why doing the set both in ram and in ems?

Reply:
WIthout doing both, Dune gameplay hangs up.

Reply:
This should not be done. Line removed. Thanks !